src/storage-types/src/sources/mysql.rs

```diff
@@ -278,15 +278,31 @@ pub struct MySqlSourceDetails {
         strategy = "proptest::collection::vec(any::<mz_mysql_util::MySqlTableDesc>(), 0..4)"
     )]
     pub tables: Vec<mz_mysql_util::MySqlTableDesc>,
-    /// The initial 'gtid_executed' set for the source. This is used as the effective
-    /// snapshot point, to ensure consistency if the source is interrupted but commits
-    /// one or more tables before the initial snapshot of all tables is complete.
+    /// The initial 'gtid_executed' set for each subsource. The index of each string in this
+    /// vector corresponds to the index of the corresponding table in the `tables` field.
+    /// If this vector has only one element, it is the initial gtid set for all tables.
+    ///
+    /// This is used as the effective snapshot point for each subsource to ensure correctness
+    /// if the source is interrupted but commits one or more tables before the initial snapshot
+    /// of all tables is complete.
     #[proptest(strategy = "any_gtidset()")]
```

Comment:
I think you want to want to change this to be a vec instead of changing the `any_gtidset` function to return a vector (or that's the route we take we should change its name).

```suggestion
    #[proptest(strategy = ""proptest::collection::vec(any_gtidset(), 1..4)")]
```